### PR TITLE
Replace deprecated context flags with NONE in benchmarks and tests 

### PR DESCRIPTION
**Pull request by jonasnick**
Based on #1126.

### CODE REVIEW

File: src/tests.c [hunk 1]
```diff
@@ -146,75 +146,89 @@ void run_selftest_tests(void) {
...
+    size_t ctx_size = secp256k1_context_preallocated_size(SECP256K1_CONTEXT_DEFAULT);
+    int i;
+
+    /* Check that SECP256K1_CONTEXT_DEFAULT is just an alias of the deprecated SECP256K1_CONTEXT_SIGN. */
+    CHECK(SECP256K1_CONTEXT_DEFAULT == SECP256K1_CONTEXT_SIGN);
+
+    /* Check that a context created with any of the flags in the flags array is
+     * identical to the DEFAULT context. We use the preallocated interface to
+     * allow setting the memory area before creating the context and therefore
+     * to ignore the padding when comparing the contexts. */
```
[the remark this answers is not in the document]
Comment (jonasnick) [reply src/tests.c]: Thanks.... Added a function that compares the members.

File: src/tests.c [hunk 2]
```diff
@@ -146,22 +146,52 @@ void run_selftest_tests(void) {
...
+        || a->illegal_callback.fn != b->illegal_callback.fn
+        || a->illegal_callback.data != b->illegal_callback.data
+        || a->error_callback.fn != b->error_callback.fn
+        || a->error_callback.data != b->error_callback.data) {
+        return 0;
+    }
+    return 1;
+}
+
+void test_deprecated_flags(void) {
```
Comment (real-or-random) [inline src/tests.c:170]: This could also test consistency of `secp256k1_context_preallocated_clone_size`

**Review by real-or-random — COMMENTED** [review]
Concept ACK

File: src/tests.c (hunk 2 above)
Comment (jonasnick) [reply src/tests.c:170]: I'm not sure how it relates to deprecated flags. If the contexts are all equal than the size of their clones is equal.

File: src/tests.c [hunk 3]
```diff
@@ -3872,6 +3872,22 @@ void run_gej(void) {
...
+    /* Tests for secp256k1_gej_eq_var */
+    for (i = 0; i < count; i++) {
+        secp256k1_fe fe;
+        random_gej_test(&a);
+        random_gej_test(&b);
+        CHECK(!secp256k1_gej_eq_var(&a, &b));
+
+        b = a;
+        random_field_element_test(&fe);
+        if(secp256k1_fe_is_zero(&fe)) {
```
[the remark this answers is not in the document]
Comment (jonasnick) [reply src/tests.c:3863]: fixed

File: src/tests.c [hunk 4]
```diff
@@ -146,22 +146,52 @@ void run_selftest_tests(void) {
     secp256k1_selftest();
 }
 
+int ecmult_gen_context_eq(const secp256k1_ecmult_gen_context *a, const secp256k1_ecmult_gen_context *b) {
+    if (a->built != b->built
+        || !secp256k1_scalar_eq(&a->blind, &b->blind)
+        || !secp256k1_gej_eq_var(&a->initial, &b->initial)) {
+        return 0;
+    }
+    return 1;
```
[the remark this answers is not in the document]
Comment (jonasnick) [reply src/tests.c:155]: An AI wrote this code for me. I did not review it. Fixed now.

File: src/bench_internal.c [hunk 5]
```diff
@@ -395,8 +387,7 @@ int main(int argc, char **argv) {
     if (d || have_flag(argc, argv, "hash") || have_flag(argc, argv, "hmac")) run_benchmark("hash_hmac_sha256", bench_hmac_sha256, bench_setup, NULL, &data, 10, iters);
     if (d || have_flag(argc, argv, "hash") || have_flag(argc, argv, "rng6979")) run_benchmark("hash_rfc6979_hmac_sha256", bench_rfc6979_hmac_sha256, bench_setup, NULL, &data, 10, iters);
 
-    if (d || have_flag(argc, argv, "context") || have_flag(argc, argv, "verify")) run_benchmark("context_verify", bench_context_verify, bench_setup, NULL, &data, 10, 1 + iters/1000);
-    if (d || have_flag(argc, argv, "context") || have_flag(argc, argv, "sign")) run_benchmark("context_sign", bench_context_sign, bench_setup, NULL, &data, 10, 1 + iters/100);
+    if (d || have_flag(argc, argv, "context")) run_benchmark("context_create", bench_context, bench_setup, NULL, &data, 10, 1 + iters/1000);
```
Comment (real-or-random) [inline src/bench_internal.c:390]: nit: I think `iters` is better than `iters/1000` now that context creation is very fast

File: src/tests.c [hunk 6]
```diff
@@ -311,49 +296,38 @@ void run_context_tests(int use_prealloc) {
...
+    } while(!secp256k1_ecdsa_sig_sign(&ctx->ecmult_gen_ctx, &sigr, &sigs, &key, &msg, &nonce, NULL));
 
     /* try signing */
-    CHECK(secp256k1_ecdsa_sig_sign(&sign->ecmult_gen_ctx, &sigr, &sigs, &key, &msg, &nonce, NULL));
-    CHECK(secp256k1_ecdsa_sig_sign(&both->ecmult_gen_ctx, &sigr, &sigs, &key, &msg, &nonce, NULL));
+    CHECK(secp256k1_ecdsa_sig_sign(&ctx->ecmult_gen_ctx, &sigr, &sigs, &key, &msg, &nonce, NULL));
 
     /* try verifying */
     CHECK(secp256k1_ecdsa_sig_verify(&sigr, &sigs, &pub, &msg));
     CHECK(secp256k1_ecdsa_sig_verify(&sigr, &sigs, &pub, &msg));
```
Comment (real-or-random) [inline src/tests.c:316]: now that you touch this, maybe remove the duplicate line

**Review by real-or-random — COMMENTED** [review]
ACK mod nits

File: src/secp256k1.c [hunk 7]
```diff
@@ -57,6 +57,8 @@
     } \
 } while(0)
 
+/* Note that whenever you change the context struct, you must also change the
+ * context_eq function. */
```
Comment (real-or-random) [inline src/secp256k1.c:61]: maybe add "in the tests"

File: src/bench_internal.c (hunk 5 above)
Comment (jonasnick) [reply src/bench_internal.c]: makes sense, done

File: src/tests.c (hunk 6 above)
Comment (jonasnick) [reply src/tests.c]: done

File: src/secp256k1.c (hunk 7 above)
Comment (jonasnick) [reply src/secp256k1.c:61]: My thinking was that if we ever move this then it's unlikely we update this line. So just having someone searching it is more robust.

Comment (real-or-random) [reply src/secp256k1.c:61]: Agreed.

File: src/modules/extrakeys/tests_impl.h [hunk 8]
```diff
@@ -260,33 +245,32 @@ void test_xonly_pubkey_tweak_check(void) {
     unsigned char tweak[32];
 
     int ecount;
-    secp256k1_context *none = api_test_context(SECP256K1_CONTEXT_NONE, &ecount);
-    secp256k1_context *sign = api_test_context(SECP256K1_CONTEXT_SIGN, &ecount);
-    secp256k1_context *verify = api_test_context(SECP256K1_CONTEXT_VERIFY, &ecount);
+
+    set_counting_callbacks(ctx, &ecount);
```
Comment (real-or-random) [inline src/modules/extrakeys/tests_impl.h:249]: Here's an example of #1167 in the wild. (maybe that's how you found this issue?)

Anyway, we have this pattern many times in the tests. This needs separate solution that should not hold off this PR.

**Review by real-or-random — APPROVED** [review]
ACK d6dc0f4ae33d3cd25e9731b9d63b4a34600bc535 diff looks good and tests pass locally

File: src/tests.c [hunk 9]
```diff
@@ -146,22 +146,47 @@ void run_selftest_tests(void) {
...
+            && secp256k1_scalar_eq(&a->blind, &b->blind)
+            && secp256k1_gej_eq_var(&a->initial, &b->initial);
+}
+
+int context_eq(const secp256k1_context *a, const secp256k1_context *b) {
+    return a->declassify == b->declassify
+            && ecmult_gen_context_eq(&a->ecmult_gen_ctx, &b->ecmult_gen_ctx)
+            && a->illegal_callback.fn == b->illegal_callback.fn
+            && a->illegal_callback.data == b->illegal_callback.
+data
```
[the remark this answers is not in the document]
Comment (real-or-random) [reply src/tests.c:160]: I'm about to open a follow up, let me fix it there

**Pull request merged**